File: android/app/src/main/kotlin/com/wagmilabs/sigil/ui/pairing/AwaitingAdminApprovalView.kt
package com.wagmilabs.sigil.ui.pairing

import androidx.compose.foundation.layout.*
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.AccessTime
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.res.stringResource
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.ui.unit.dp
import com.wagmilabs.sigil.R

/**
 * OOB approval waiting screen - displayed while polling /pair/complete
 * Shows when server returns NOT_APPROVED error
 *
 * Mirror of iOS AwaitingAdminApprovalView per Phase 3
 */
@Composable
fun AwaitingAdminApprovalView(
    serverId: String,
    pollAttempts: Int,
    modifier: Modifier = Modifier
) {
    val timeElapsed = remember(pollAttempts) {
        val seconds = pollAttempts * 2
        val minutes = seconds / 60
        val remainingSeconds = seconds % 60

        if (minutes > 0) {
            "${minutes}m ${remainingSeconds}s"
        } else {
            "${seconds}s"
        }
    }

    val pollProgress = remember(pollAttempts) {
        val maxAttempts = 150f // 5 minutes at 2s intervals
        pollAttempts / maxAttempts
    }

    Column(
        modifier = modifier
            .fillMaxSize()
            .padding(24.dp),
        horizontalAlignment = Alignment.CenterHorizontally,
        verticalArrangement = Arrangement.spacedBy(24.dp)
    ) {
        Spacer(modifier = Modifier.weight(0.2f))

        // Spinner
        CircularProgressIndicator(
            modifier = Modifier.size(48.dp)
        )

        // Title
        Text(
            text = stringResource(R.string.pair_oob_title),
            style = MaterialTheme.typography.titleLarge,
            fontWeight = FontWeight.SemiBold,
            textAlign = TextAlign.Center
        )

        // Subtitle
        Text(
            text = stringResource(R.string.pair_oob_subtitle),
            style = MaterialTheme.typography.bodyMedium,
            color = MaterialTheme.colorScheme.onSurfaceVariant,
            textAlign = TextAlign.Center,
            modifier = Modifier.padding(horizontal = 16.dp)
        )

        // Server info card
        Card(
            modifier = Modifier.fillMaxWidth()
        ) {
            Column(
                modifier = Modifier.padding(16.dp),
                verticalArrangement = Arrangement.spacedBy(8.dp)
            ) {
                Text(
                    text = stringResource(R.string.pair_oob_server_label),
                    style = MaterialTheme.typography.labelSmall,
                    color = MaterialTheme.colorScheme.onSurfaceVariant
                )
                Text(
                    text = serverId,
                    style = MaterialTheme.typography.bodyLarge,
                    fontWeight = FontWeight.Medium
                )
            }
        }

        // Status info
        Column(
            horizontalAlignment = Alignment.CenterHorizontally,
            verticalArrangement = Arrangement.spacedBy(8.dp)
        ) {
            Row(
                horizontalArrangement = Arrangement.spacedBy(8.dp),
                verticalAlignment = Alignment.CenterVertically
            ) {
                Icon(
                    imageVector = Icons.Default.AccessTime,
                    contentDescription = null,
                    tint = MaterialTheme.colorScheme.onSurfaceVariant,
                    modifier = Modifier.size(16.dp)
                )
                Text(
                    text = stringResource(R.string.pair_oob_time_elapsed, timeElapsed),
                    style = MaterialTheme.typography.labelMedium,
                    color = MaterialTheme.colorScheme.onSurfaceVariant
                )
            }

            LinearProgressIndicator(
                progress = pollProgress,
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(horizontal = 16.dp)
            )

            Text(
                text = stringResource(R.string.pair_oob_poll_status),
                style = MaterialTheme.typography.labelSmall,
                color = MaterialTheme.colorScheme.onSurfaceVariant
            )
        }

        Spacer(modifier = Modifier.weight(1f))

        // Info text
        Text(
            text = stringResource(R.string.pair_oob_wait_message),
            style = MaterialTheme.typography.bodySmall,
            color = MaterialTheme.colorScheme.onSurfaceVariant,
            textAlign = TextAlign.Center,
            modifier = Modifier.padding(horizontal = 16.dp)
        )

        Spacer(modifier = Modifier.height(16.dp))
    }
}

@Preview(showBackground = true)
@Composable
fun AwaitingAdminApprovalViewPreview() {
    MaterialTheme {
        AwaitingAdminApprovalView(
            serverId = "sigil-prod-01",
            pollAttempts = 15
        )
    }
}

File: ios/Sources/UI/Pairing/AwaitingAdminApprovalView.swift
import SwiftUI

/// OOB approval waiting screen - displayed while polling /pair/complete
/// Shows when server returns NOT_APPROVED error
struct AwaitingAdminApprovalView: View {
    let serverId: String
    let pollAttempts: Int

    private var timeElapsed: String {
        let seconds = pollAttempts * 2
        let minutes = seconds / 60
        let remainingSeconds = seconds % 60

        if minutes > 0 {
            return "\(minutes)m \(remainingSeconds)s"
        } else {
            return "\(seconds)s"
        }
    }

    private var pollProgress: Double {
        let maxAttempts = 150.0 // 5 minutes at 2s intervals
        return Double(pollAttempts) / maxAttempts
    }

    var body: some View {
        VStack(spacing: 24) {
            // Spinner
            ProgressView()
                .scaleEffect(1.5)
                .padding(.bottom, 8)

            // Title
            Text.localized("pair-oob-title")
                .font(.title2)
                .fontWeight(.semibold)
                .multilineTextAlignment(.center)

            // Subtitle
            Text.localized("pair-oob-subtitle")
                .font(.body)
                .foregroundColor(.secondary)
                .multilineTextAlignment(.center)
                .padding(.horizontal)

            // Server info card
            VStack(alignment: .leading, spacing: 8) {
                Text.localized("pair-oob-server-label")
                    .font(.caption)
                    .foregroundColor(.secondary)
                Text(serverId)
                    .font(.body)
                    .fontWeight(.medium)
            }
            .frame(maxWidth: .infinity, alignment: .leading)
            .padding()
            .background(Color(.systemGray6))
            .cornerRadius(8)
            .padding(.horizontal)

            // Status info
            VStack(spacing: 8) {
                HStack {
                    Image(systemName: "clock")
                        .foregroundColor(.secondary)
                    Text(LocalizationService.shared.string("pair-oob-time-elapsed", arguments: ["timeElapsed": .string(timeElapsed)]))
                        .font(.caption)
                        .foregroundColor(.secondary)
                }

                ProgressView(value: pollProgress)
                    .progressViewStyle(.linear)
                    .padding(.horizontal)

                Text.localized("pair-oob-poll-status")
                    .font(.caption2)
                    .foregroundColor(.secondary)
            }
            .padding(.horizontal)

            Spacer()

            // Info text
            Text.localized("pair-oob-wait-message")
                .font(.caption)
                .foregroundColor(.secondary)
                .multilineTextAlignment(.center)
                .padding(.horizontal)
                .padding(.bottom)
        }
        .padding()
    }
}

#Preview {
    AwaitingAdminApprovalView(
        serverId: "sigil-prod-01",
        pollAttempts: 15
    )
}
